From b9d88e52f8c1a4a27daab5739e6c777f2dbb655a Mon Sep 17 00:00:00 2001
Date: Tue, 26 Jun 2018 15:12:48 +0000
Subject: [PATCH] Add FPDFTextObj_GetFontSize() API
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In contrast with FPDFText_GetFontSize(), this exposes the font size of
the text object according to the text state, rather than the font size
of a particular character.

Change-Id: Iac88d1aea8fb6bb5522bdaf01363aa6d32025b8f
Reviewed-on: https://pdfium-review.googlesource.com/35931
Reviewed-by: Nicol치s Pe침a Moreno <npm@chromium.org>
Commit-Queue: Nicol치s Pe침a Moreno <npm@chromium.org>
---
 fpdfsdk/fpdf_edit_embeddertest.cpp |  3 +++
 fpdfsdk/fpdf_edittext.cpp          | 11 +++++++++++
 fpdfsdk/fpdf_view_c_api_test.c     |  1 +
 public/fpdf_edit.h                 |  9 +++++++++
 4 files changed, 24 insertions(+)

diff --git a/fpdfsdk/fpdf_edittext.cpp b/fpdfsdk/fpdf_edittext.cpp
index a927e16e1..e339c2412 100644
--- a/fpdfsdk/fpdf_edittext.cpp
+++ b/fpdfsdk/fpdf_edittext.cpp
@@ -511,6 +511,17 @@ FPDF_EXPORT FPDF_BOOL FPDF_CALLCONV FPDFText_GetMatrix(FPDF_PAGEOBJECT text,
   return true;
 }
 
+FPDF_EXPORT double FPDF_CALLCONV FPDFTextObj_GetFontSize(FPDF_PAGEOBJECT text) {
+  if (!text)
+    return 0;
+
+  CPDF_TextObject* pTextObj = CPDFTextObjectFromFPDFPageObject(text);
+  if (!pTextObj)
+    return 0;
+
+  return pTextObj->GetFontSize();
+}
+
 FPDF_EXPORT void FPDF_CALLCONV FPDFFont_Close(FPDF_FONT font) {
   CPDF_Font* pFont = CPDFFontFromFPDFFont(font);
   if (!pFont)
diff --git a/public/fpdf_edit.h b/public/fpdf_edit.h
index 6df5e3237..6e613bca0 100644
--- a/public/fpdf_edit.h
+++ b/public/fpdf_edit.h
@@ -1090,6 +1090,15 @@ FPDF_EXPORT FPDF_BOOL FPDF_CALLCONV FPDFText_GetMatrix(FPDF_PAGEOBJECT text,
                                                        double* e,
                                                        double* f);
 
+// Experimental API.
+// Get the font size of a text object.
+//
+//   text - handle to a text.
+//
+// Returns the font size of the text object, measured in points (about 1/72
+// inch) on success; 0 on failure.
+FPDF_EXPORT double FPDF_CALLCONV FPDFTextObj_GetFontSize(FPDF_PAGEOBJECT text);
+
 // Close a loaded PDF font.
 //
 // font   - Handle to the loaded font.
-- 
2.16.4

